client/desktop/lib/keychain/index.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * External dependencies
+ */
+const keytar = require( 'keytar' );
+
+/*
+ * Module constants
+ */
+const keychainService = 'WordPress.com';
+const credentialsKey = 'User Credentials';
+
+function set( key, value ) {
+	return keytar.setPassword( keychainService, key, value );
+}
+
+function get( key ) {
+	return keytar.getPassword( keychainService, key );
+}
+
+async function setUserInfo( info ) {
+	return await set( credentialsKey, JSON.stringify( info ) );
+}
+
+async function getUserInfo() {
+	const info = await get( credentialsKey );
+	if ( info ) {
+		return JSON.parse( info );
+	}
+	return null;
+}
+
+async function clear() {
+	const credentials = await keytar.findCredentials( keychainService );
+	if ( credentials && Array.isArray( credentials ) && credentials.length > 0 ) {
+		for ( let i = 0; i < credentials.length; i++ ) {
```

Comment:
Could you use credentials.forEach here?

Reply:
Done in 66897dde8a4a885cd908f31233218bd0fcba64e4